app/views/books/index.html.erb

```diff
@@ -1,20 +1,22 @@
+<%= turbo_stream_from :books %>
+
 <div class="container">
   <h1 class="text-center mb-5">Books</h1>
-  <div class="row">
-    <% @books.each do |book| %>
-      <div class="col-md-6">
-        <div class="book">
-          <h2 class="book-title"><%= book.title %></h2>
-          <div class="book-details">by <%= book.author.full_name %></div>
-          <div class="book-details">Published in <%= book.year %></div>
-          <div class="book-category">Category: <%= book.category.name %></div>
-          <div class="book-btns">
-            <%= link_to 'Details', book_path(book), class: "btn book-btn btn-details" %>
-            <%= reserve_book_button(book, css_class: 'book-btn btn-reserve') %>
-            <%= loan_book_button(book, css_class: 'book-btn btn-reserve') %>
-          </div>
-        </div>
+  <div class='text-center mb-5' data-controller='search'>
+    <%= text_field_tag :query,'', placeholder: 'Search by title', data: { action: 'input->search#connect', search_target: :params } %>
```

Comment:
tutaj też taki pro tip; bo to dziala:) i wiem ze tak było w instrukcji (troche zeby nie wprowadzac za dużo ;) ), ale ;D

`connect` jest domyślną funkcją która, odpala się w momencie podpięcia stimulusa:
`Anytime the controller is connected to the DOM` (zrodlo: https://stimulus.hotwired.dev/reference/lifecycle-callbacks )

Tak naprawdę - ja bym była za tym, zeby napisać swoją, i podpinać swoją :)
Sprowadza się to do tego, że np tutaj podpinasz

 `data: { action: 'input->search#find', search_target: :params }`
i w kontrolerze stimulusowym w `find` obslugujesz:

```javascript
export default class extends Controller {
  static targets = ["params"]

  find() {
    const value = this.paramsTarget.value
```